### PR TITLE
[JENKINS-27414] Don't offer to restart if unsupported

### PR DESCRIPTION
**Pull request by daniel-beck**

### CODE REVIEW

**Comment by olivergondza** [comment]
:+1:

**Comment by daniel-beck** [comment]
I wonder whether additionally showing the exception message of `Lifecycle.verifyRestartable()` would be helpful here. (Just the message, not the stack trace.)

File: core/src/main/resources/jenkins/model/Jenkins/_restart.jelly [hunk 1]
```diff
@@ -28,10 +28,17 @@ THE SOFTWARE.
...
-		  </form>
+			<j:choose>
+				<j:when test="${app.lifecycle.canRestart()}">
+					<form method="post" action="restart">
+						${%Are you sure about restarting Jenkins?}
+						<f:submit value="${%Yes}" />
+					</form>
+				</j:when>
+				<j:otherwise>
+					${%Jenkins cannot restart itself as currently configured.}
```
Comment (oleg-nenashev) [inline core/src/main/resources/jenkins/model/Jenkins/_restart.jelly]: Minor comment.
I would move this phrase to the localization dictionary. It's too long for a key IMO. However, there's already a sentence ("Are you sure...") being used in such way, so probably there's no extra problems

Comment (daniel-beck) [inline core/src/main/resources/jenkins/model/Jenkins/_restart.jelly]: Right, that's the reason I did it like this.

**Comment by KostyaSha** [comment]
I don't remember wether cli has restart command, but will it work for cli requests?

**Comment by daniel-beck** [comment]
@KostyaSha I don't know whether there's a way for the `restart` method to be conditional on Jenkins being restartable. It throws a `RestartNotSupported` exception, but the CLI generally suffers from being much closer to internals like that.

Note that this is only cosmetic and aims to prevent user confusion and the stack trace from showing on the UI.

**Pull request merged**